control-plane-operator/controllers/hostedcontrolplane/creatorupdate_ownerref_enforcer.go

```diff
@@ -2,18 +2,57 @@ package hostedcontrolplane
 
 import (
 	"context"
+	"fmt"
 
+	configv1 "github.com/openshift/api/config/v1"
 	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
 	"github.com/openshift/hypershift/support/config"
 	"github.com/openshift/hypershift/support/upsert"
+	"github.com/openshift/hypershift/support/util"
+	prometheusoperatorv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
 	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+func retrieveHostedCLuster(ctx context.Context, c crclient.Client, hclusterNamespacedName string) *hyperv1.HostedCluster {
+	hcluster := &hyperv1.HostedCluster{}
+
+	err := c.Get(ctx, util.ParseNamespacedName(hclusterNamespacedName), hcluster)
+	if err != nil {
+		fmt.Printf("failed to get cluster %q: %s\n", hclusterNamespacedName, err)
+
+		return nil
+	}
+
+	return hcluster
+}
+
 func createOrUpdateWithOwnerRefFactory(upstream upsert.CreateOrUpdateFN) func(*hyperv1.HostedControlPlane) upsert.CreateOrUpdateFN {
```

Comment:
Ok, I will add a dedicated function conditionally calling `createOrUpdate`; this will also avoid having the switch on the type. To be honest I was quite expecting this comment.

Reply:
Indeed this makes sense, I will move the code in a dedicated wrapper.